diff --git a/opensnitch/utils/__init__.py b/opensnitch/utils/__init__.py
index ce5ae608..2000c0d6 100644
--- a/opensnitch/utils/__init__.py
+++ b/opensnitch/utils/__init__.py
@@ -283,8 +283,8 @@ class QuickHelp():
 class Utils():
     @staticmethod
     def check_versions(daemon_version):
-        lMayor, lMinor, lPatch = gui_version.split(".")
-        rMayor, rMinor, rPatch = daemon_version.split(".")
+        lMayor, lMinor, lPatch = gui_version.split(".")[:3]
+        rMayor, rMinor, rPatch = daemon_version.split(".")[:3]
         return lMayor != rMayor or (lMayor == rMayor and lMinor != rMinor)
 
     @staticmethod
