diff --git a/audisp/audispd.c b/audisp/audispd.c
index 2951aad2..ee6282fa 100644
--- a/audisp/audispd.c
+++ b/audisp/audispd.c
@@ -36,6 +36,7 @@
 #include <arpa/inet.h>
 #include <limits.h>
 #include <sys/uio.h>
+#include <sys/stat.h>
 #include <getopt.h>
 
 #include "audispd-pconfig.h"
@@ -112,15 +113,30 @@ static void load_plugin_conf(conf_llist *plugin)
 	d = opendir(daemon_config.plugin_dir);
 	if (d) {
 		struct dirent *e;
+		struct stat *s;
 
 		while ((e = readdir(d))) {
 			plugin_conf_t config;
 			char fname[PATH_MAX];
+			char *rp;
 			const char *ext, *reason = NULL;
 
-			if (e->d_type != DT_REG)
-				reason = "not a regular file";
-			else if (e->d_name[0] == '.')
+			snprintf(fname, sizeof(fname), "%s/%s",
+				daemon_config.plugin_dir, e->d_name);
+
+			if (e->d_type == DT_LNK) {
+				rp = realpath(fname, NULL); // follow symlinks, use realpath to do plugin logic instead
+			} else {
+				rp = strdup(fname); // can't just set pointer using =, or free becomes undefined behavior
+			}
+
+			if (rp == NULL)
+				reason = "unable to allocate buffer";
+			else if (lstat(rp, s) < 0) // need to check file type on the absolute path, which requires stat
+				reason = "unable to stat";
+			else if (!S_ISREG(s->st_mode))
+				reason = "not a regular file or symlink";
+			else if (e->d_name[0] == '.') // name logic happens on the actual entry in the plugin directory
 				reason = "hidden file";
 			else if (count_dots(e->d_name) > 1)
 				reason = "backup file";
@@ -132,11 +148,8 @@ static void load_plugin_conf(conf_llist *plugin)
 				continue;
 			}
 
-			snprintf(fname, sizeof(fname), "%s/%s",
-				daemon_config.plugin_dir, e->d_name);
-
 			clear_pconfig(&config);
-			if (load_pconfig(&config, fname) == 0) {
+			if (load_pconfig(&config, rp) == 0) {
 				/* Push onto config list only if active */
 				if (config.active == A_YES)
 					plist_append(plugin, &config);
@@ -146,6 +159,8 @@ static void load_plugin_conf(conf_llist *plugin)
 				audit_msg(LOG_ERR,
 					"Skipping %s plugin due to errors",
 					e->d_name);
+
+			free(rp);
 		}
 		closedir(d);
 	}
