diff --git a/lib/message.c b/lib/message.c
index 2e20f76b..c68a7456 100644
--- a/lib/message.c
+++ b/lib/message.c
@@ -21,42 +21,40 @@
  */
 
 #include "config.h"
-#include <errno.h>
-#include <stdio.h>
-#include <stdarg.h>
 #include "libaudit.h"
 #include "private.h"
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
 
 /* The message mode refers to where informational messages go
    0 - stderr, 1 - syslog, 2 - quiet. The default is quiet. */
-static message_t message_mode = MSG_QUIET;
-static debug_message_t debug_message = DBG_NO;
+static message_t message_mode = MSG_STDERR;
+static debug_message_t debug_message = DBG_YES;
 
-void set_aumessage_mode(message_t mode, debug_message_t debug)
-{
-        message_mode = mode;
-	debug_message = debug;
+void set_aumessage_mode(message_t mode, debug_message_t debug) {
+  message_mode = mode;
+  debug_message = debug;
 }
 
-void audit_msg(int priority, const char *fmt, ...)
-{
-        va_list   ap;
-        int saved_errno = errno;
+void audit_msg(int priority, const char *fmt, ...) {
+  va_list ap;
+  int saved_errno = errno;
 
-	if (message_mode == MSG_QUIET)
-		return;
+  if (message_mode == MSG_QUIET)
+    return;
 
-	if (priority == LOG_DEBUG && debug_message == DBG_NO)
-		return;
+  if (priority == LOG_DEBUG && debug_message == DBG_NO)
+    return;
 
-        va_start(ap, fmt);
-        if (message_mode == MSG_SYSLOG)
-                vsyslog(priority, fmt, ap);
-        else {
-                vfprintf(stderr, fmt, ap);
-		fputc('\n', stderr);
-	}
-        va_end( ap );
+  va_start(ap, fmt);
+  if (message_mode == MSG_SYSLOG)
+    vsyslog(priority, fmt, ap);
+  else {
+    vfprintf(stderr, fmt, ap);
+    fputc('\n', stderr);
+  }
+  va_end(ap);
 
-        errno = saved_errno;
+  errno = saved_errno;
 }
